### PR TITLE
Elimination of redundant memory access and calculations in cell capture

### PR DESCRIPTION
**Pull request by MattRolchigo**
Renamed variables and added comments to make cell capture subroutine more descriptive, and removed/optimized calculation of certain intermediate quantities. Up to a 5% performance improvement by reducing the amount of memory access (particularly for `diagonal_length` and `cell_type`)

### CODE REVIEW

File: src/CAupdate.hpp [hunk 1]
```diff
@@ -313,68 +312,80 @@ void cellCapture(const int, const int np, const Grid &grid, const InterfacialRes
...
+                                    proj_next_nearest_corner_edge_2 = dist_second_corner - proj_nearest_corner_edge_2;
                                 }
-                                float l_12 = 0.5 * (Kokkos::fmin(i_1, Kokkos::sqrt(3.0f)) +
-                                                    Kokkos::fmin(i_2, Kokkos::sqrt(3.0f)));
-                                float l_13 = 0.5 * (Kokkos::fmin(j_1, Kokkos::sqrt(3.0f)) +
-                                                    Kokkos::fmin(j_2, Kokkos::sqrt(3.0f)));
+
+                                // Truncate the lengths at sqrt(3) for a max initial size of an octahedron
+                                const float l_12 =
+                                    0.5 * (Kokkos::fmin(proj_nearest_corner_edge_1, Kokkos::sqrt(3.0f)) +
```
Comment (streeve) [inline src/CAupdate.hpp:368]: Could make `sqrt(3)` a class constant or cache it above

**Review by streeve — COMMENTED** [review]
All seems reasonable, but also very easy place to introduce a bug. Any thoughts (new test, throw caution to the wind, etc.)?

**Comment by MattRolchigo** [comment]
Luckily we do have a test for this - CAupdate would've failed for the single grain or the directional solidification problem if we messed up (which I did do at one point), but cell capture seems to be working at intended based on the number of time steps it takes to complete the small example problems. I'll definitely want to test a couple large problems on GPU to make sure no weird floating point edge case got introduced though

File: src/CAupdate.hpp (hunk 1 above)
Comment (MattRolchigo) [reply src/CAupdate.hpp:368]: I thought about doing that, I've seen that done in some older CPU codes before. I figured the memory access of the constant vs the time to compute sqrt(3) would be pretty comparable, but worth checking

Comment (MattRolchigo) [reply src/CAupdate.hpp:368]: It doesn't seem to have made a difference for performance either way, so I'll just leave it as is for simplicity

**Review by streeve — APPROVED** [review]
